### PR TITLE
Core only dockerfile

### PR DESCRIPTION
**Pull request by wochinge**
**Motivation**:
Issue: https://github.com/RasaHQ/rasa_core/issues/845

**Proposed changes**:
- remove nlu from rasa core Dockerfile
- add docker compose file which straps together core, nlu, core sdk and duckling

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] updated the changelog

### CODE REVIEW

**Comment by wochinge** [comment]
Ok, I add the documentation.
Should we add an extra volume in the `Dockerfile` for the credentials?

**Comment by tmbo** [comment]
I mean there is already a mount for `endpoints` - they are in the same file. so we might consider renaming the `credentials.yml` to something more reasonable in the documentation, e.g. `endpoints.yml`

**Comment by wochinge** [comment]
But `credentials.yml` somehow makes sense as it is specified by `--credentials`, doesn't it?

**Comment by wochinge** [comment]
This is kind of escalating quickly 😄 but how about adding another volume so that users can add their project directory (training data, domain.yml, ...)? 
We could also make a volume `config` which can both contain `endpoints.yml` and `credentials.yml`. I think there is no real need to keep this in separate directories.

**Comment by tmbo** [comment]
yes I agree, let's assume that these files are in the same directory

**Comment by tmbo** [comment]
yes I agree, let's assume that these files are in the same directory

**Comment by wochinge** [comment]
@tmbo ready for another review (commit might not yet be visible in Github due to Github's restore operations)

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+version: '3.0'
+
+services:
+  rasacore:
+      image: rasa/rasa_core:latest
+      networks: ['rasa-network']
+      ports:
+      - "5005:5005"
+      volumes:
+      - "./rasa-app-data/core-model:/app/model"
```
Comment (tmbo) [inline docker/docker-compose.yml]: we do have a starter pack to get run from when creating a new bot, can you please make sure the docker compose and that repository play well with one another https://github.com/RasaHQ/starter-pack-rasa-stack

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+version: '3.0'
+
+services:
+  rasacore:
```
Comment (tmbo) [inline docker/docker-compose.yml]: i think we should stick to one naming convention (`action_serve` vs `rasacore`), i guess this should be `rasa_core`

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        - --endpoints
+        - config/endpoints.yml
+  action_server:
+      image: rasa/rasa_core_sdk:latest
+      networks: ['rasa-network']
+      ports:
+      - "5055:5055"
+      volumes:
+      - "./rasa-app-data/actions:/app/actions"
+  rasanlu:
```
Comment (tmbo) [inline docker/docker-compose.yml]: `rasa_nlu`

File: docs/installation.rst [hunk 4]
```diff
@@ -87,14 +90,82 @@ Rasa Stack starter-pack which has all the files you need to build your first cus
...
+
+.. note::
+
+    If you do not require components like `nlu <https://rasa.com/docs/nlu/>`_ or duckling,
+    you can simply remove them from your docker compose file.
+
+Running it
+~~~~~~~~~~
+To run all components locally, execute :code:`docker-compose up`.
+You can then interact with your chat bot using the `Rasa Core HTTP API <https://rasa.com/docs/core/server/>`_.
```
Comment (tmbo) [inline docs/installation.rst]: since this is a link within the core docs, let's use 
```
:ref:`section_http` 
```

File: docs/installation.rst [hunk 5]
```diff
@@ -87,14 +90,82 @@ Rasa Stack starter-pack which has all the files you need to build your first cus
...
+    .. code-block:: bash
+
+         docker-compose run rasacore train
+
+Volume Explanation
+~~~~~~~~~~~~~~~~~~
+- **./rasa-app-data/core-model**: This directory contains the trained Rasa Core models.
+  You can also move previously trained models to this directory to load them within the Docker container.
+- **./rasa-app-data/config**: This directory is for the configuration of the endpoints and of the
+  different `chat & voice platforms <https://rasa.com/docs/core/connectors/>`_ you can use Rasa Core with.
```
Comment (tmbo) [inline docs/installation.rst]: same here, docs internal links should use `:ref:`

File: docs/installation.rst [hunk 6]
```diff
@@ -87,14 +90,82 @@ Rasa Stack starter-pack which has all the files you need to build your first cus
...
+      For the shown docker-compose example the file should look like this:
+
+        .. code-block:: yaml
+
+            action_endpoint:
+                url: 'http://action_server:5055/webhook'
+            nlu:
+                url: 'http://rasanlu:5000'
+
+    - If you use a connector to a `chat & voice platforms <https://rasa.com/docs/core/connectors/>`_
```
Comment (tmbo) [inline docs/installation.rst]: `:ref:`

**Pull request merged**